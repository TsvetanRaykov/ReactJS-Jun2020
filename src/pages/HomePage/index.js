import React, { useEffect } from 'react'
import { Typography, Paper, Avatar, Button } from '@material-ui/core'
import withStyles from '@material-ui/core/styles/withStyles'
import { Link } from 'react-router-dom'
import userService from '../../services/userService'

const styles = (theme) => ({
	main: {
		width: 'auto',
		display: 'block',
		marginLeft: theme.spacing(3),
		marginRight: theme.spacing(3),
		[theme.breakpoints.up(400 + theme.spacing(3 * 2))]: {
			width: 400,
			marginLeft: 'auto',
			marginRight: 'auto',
		},
	},
	paper: {
		marginTop: theme.spacing(8),
		display: 'flex',
		flexDirection: 'column',
		alignItems: 'center',
		padding: `${theme.spacing(2)}px ${theme.spacing(3)}px ${theme.spacing(
			3
		)}px`,
	},
	avatar: {
		margin: theme.spacing(2),
		backgroundColor: theme.palette.secondary.main,
		width: theme.spacing(7),
		height: theme.spacing(7),
		border: '0',
	},
	submit: {
		marginTop: theme.spacing(3),
	},
})

function HomePage(props) {
	const { classes } = props
	useEffect(() => {
		if (userService.getCurrentUser()) {
			props.history.replace('/dashboard')
		}
	}, [props.history])

	return (
		<main className={classes.main}>
			<Paper className={classes.paper}>
				<Avatar className={classes.avatar} src='/logo.jpg' />
				<Typography component='h1' variant='h5'>
					Welcome to Quizoom
				</Typography>
				<Button
					type='submit'
					fullWidth
					variant='contained'
					color='primary'
					component={Link}
					to='/register'
					className={classes.submit}
				>
					Register
				</Button>
				<Button
					type='submit'
					fullWidth
					variant='contained'
					color='primary'
					component={Link}
					to='/login'
					className={classes.submit}
				>
					Login
				</Button>
			</Paper>
		</main>
	)
}

export default withStyles(styles)(HomePage)
