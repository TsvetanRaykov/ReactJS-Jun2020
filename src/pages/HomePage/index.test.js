import React from 'react'
import renderer from 'react-test-renderer'
import TestWrapper from '../../utils/test-wrapper'
import HomePage from '.'

jest.mock('react', () => ({
	...jest.requireActual('react'),
	useLayoutEffect: jest.requireActual('react').useEffect,
}))

describe('<HomePage />', () => {
	it('Should match the snapshot', () => {
		const wrapper = renderer.create(
			<TestWrapper>
				<HomePage />
			</TestWrapper>
		)
		expect(wrapper).toMatchSnapshot()
	})
})
