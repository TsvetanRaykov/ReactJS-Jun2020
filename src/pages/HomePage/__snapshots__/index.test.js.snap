// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> Should match the snapshot 1`] = `
<main
  className="HomePage-main-1"
>
  <div
    className="MuiPaper-root HomePage-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circular HomePage-avatar-3"
    >
      <img
        className="MuiAvatar-img"
        src="/logo.jpg"
      />
    </div>
    <h1
      className="MuiTypography-root MuiTypography-h5"
    >
      Welcome to Quizoom
    </h1>
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiButton-root MuiButton-contained HomePage-submit-4 MuiButton-containedPrimary MuiButton-fullWidth"
      href="/register"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiButton-label"
      >
        Register
      </span>
    </a>
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiButton-root MuiButton-contained HomePage-submit-4 MuiButton-containedPrimary MuiButton-fullWidth"
      href="/login"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiButton-label"
      >
        Login
      </span>
    </a>
  </div>
</main>
`;
